modules/codegen/src/smithy4s/codegen/internals/Renderer.scala

```diff
@@ -1617,7 +1617,10 @@ private[internals] class Renderer(compilationUnit: CompilationUnit) { self =>
       line"${ref.show}.${altName.capitalize}Case.widen".write
 
     case AltTN(_, _, AltValueTN.ProductAltTN(alt)) =>
-      alt.runDefault.write
+      // The `widen` is necessary in Scala 3.
+      // Without it, there are two schemas (the struct and the union) that match the type
+      // for the conversion to Hints.Binding.
+      line"${alt.runDefault}.widen".write
```

Comment:
👍

Reply:
I agree, will give it a shot

Reply:
done, this was apparently some previously-missed case, as we do this in all the other cases (just not 0-field product structs in a union)

Reply:
also removing the widen

Reply:
I rushed too much, apparently the widen is still necessary?

Reply:
Widen is back, for Scala 2 :D